zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/adhocsubprocess/AdHocSubProcessJobWorkerTest.java

```diff
@@ -0,0 +1,106 @@
+package io.camunda.zeebe.engine.processing.adhocsubprocess;
+
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import static org.assertj.core.api.Assertions.tuple;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import io.camunda.zeebe.model.bpmn.builder.AdHocSubProcessBuilder;
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeAdHocImplementationType;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobResult;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.value.BpmnElementType;
+import io.camunda.zeebe.protocol.record.value.VariableRecordValue;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.assertj.core.api.Assertions;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class AdHocSubProcessJobWorkerTest {
+
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+
+  private static final String PROCESS_ID = "process";
+  private static final String AD_HOC_SUB_PROCESS_ELEMENT_ID = "ad-hoc";
+  private static final String JOB_TYPE = "routing-agent";
+
+  @Rule public final RecordingExporterTestWatcher watcher = new RecordingExporterTestWatcher();
+
+  private BpmnModelInstance process(final Consumer<AdHocSubProcessBuilder> modifier) {
+    return Bpmn.createExecutableProcess(PROCESS_ID)
+        .startEvent()
+        .adHocSubProcess(
+            AD_HOC_SUB_PROCESS_ELEMENT_ID,
+            adHocSubProcess -> {
+              adHocSubProcess
+                  .zeebeImplementation(ZeebeAdHocImplementationType.JOB_WORKER)
+                  .zeebeJobType(JOB_TYPE);
```

Comment:
We'll need to update this one to account for the changes done in https://github.com/camunda/camunda/pull/36650. After a rebase, it should be enough to remove this line as the implementation type will be resolved by the existence of the taskDefinition.


```suggestion
              adHocSubProcess
                  .zeebeJobType(JOB_TYPE);
```